File: apps/java/FaceEmbedderApp.java
import java.nio.file.Path;
import java.nio.file.Paths;
import java.io.IOException;

import java.util.List;
import java.util.StringJoiner;

import java.util.stream.DoubleStream;
import java.util.stream.IntStream;
import java.util.stream.Collectors;
import java.util.function.Supplier;

import org.sertiscorp.oneml.face.*;

public class FaceEmbedderApp {

    static String joinAsString(Iterable<Double> it) {
        StringJoiner joiner = new StringJoiner(", ", "[", "]");
        for (Double i: it) {
            joiner.add(String.format("%.6f", i));
        }

        return joiner.toString();
    }

    public static void main(String[] args) throws IOException {
        Path basePath = Paths.get("", "../../assets/images");
        String path = basePath + "/register-set/Colin_Powell/colin_powell_0074.jpg";

        LicenseManager manager = new LicenseManager();
        manager.activateTrial();
        FaceEmbedder embedder = new FaceEmbedder(manager);
        Utils utils = new Utils(manager);

        Image img = utils.readImageCV(path);
        FaceEmbedderResult result = embedder.embed(img);

        System.out.println("Embedding size: " + Face.getEmbeddingSize());

        float[] embedding = result.getEmbedding();
        Supplier<DoubleStream> supplier = () -> IntStream.range(0, embedding.length).mapToDouble(i -> embedding[i]);
        List<Double> sample = supplier.get().limit(5).boxed().collect(Collectors.toList());

        System.out.println("Embedding sample: " + joinAsString(sample));
        System.out.println(String.format("Embedding sum: %.5f", supplier.get().sum()));

        UsageReport report = embedder.getUsage();
        report.toLog();
    }
}

File: apps/csharp/FaceEmbedderApp.cs
using OneML.Face;

class FaceEmbedderApp {
    static void PrintEmbedding(FaceEmbedderResult result) {
        Embedding emb = result.GetEmbedding();
        Console.WriteLine("Embedding size: " + emb.Count());

        string precision = "F6";
        string embSample = "[";
        float embSum = 0;
        for (int i = 0; i < emb.Count(); i++) {
            if (i < 5) {
                embSample += emb[i].ToString(precision) + ", ";
            }
            embSum += emb[i];
        }
        embSample = embSample.Substring(0, embSample.Length-2) + "]";

        Console.WriteLine("Embedding sample: " + embSample);
        Console.WriteLine("Embedding sum: " + embSum.ToString("F4"));
        Console.WriteLine("status: " + result.GetReturnStatus());
    }

    static void Main(string[] args) {
        LicenseManager manager = new LicenseManager();
        manager.ActivateTrial();

        using (FaceEmbedder embedder = new FaceEmbedder(manager))
        using (Utils utils = new Utils(manager))
        {
            string basePath = @"../../assets/images/";
            string path = basePath + @"register-set/Colin_Powell/colin_powell_0074.jpg";
            Image img = utils.ReadImageCV(path);
            FaceEmbedderResult result1 = embedder.Embed(img);
            Console.WriteLine("Original Embedding");
            PrintEmbedding(result1);

            FaceEmbedderResult result2 = embedder.Embed(img, true);
            Console.WriteLine("Flipped Image Embedding");
            PrintEmbedding(result2);

        }
    }
}